src/Resources/ResourceManager/Implementation/Policy/NewAzurePolicyAssignment.cs

```diff
@@ -194,15 +208,55 @@ private string GetResourceId()
             return this.GetPolicyArtifactFullyQualifiedId(this.Scope, Constants.MicrosoftAuthorizationPolicyAssignmentType, this.Name);
         }
 
+        /// <summary>
+        /// Verifies if identity type and/or identity ID and/or location are present
+        /// </summary>
+        private bool CheckIfIdentityPresent()
+        {
+            if (this.IdentityType != null & this.Location == null || this.AssignIdentity != null && this.AssignIdentity.IsPresent && this.Location == null)
```

Comment:
Could you throw invalid operation when the user specified both identityType and assignIdentity? so you don't need to deal the conflict when the user specified identityType userAssigned and assignIdentity at the same time.

Reply:
And also check whether the user is using AssignIdentity and identityId at the same time.

Reply:
Fixed